x-pack/plugins/security_solution/server/lib/detection_engine/rules/saved_object_mappings.ts

```diff
@@ -47,11 +53,30 @@ export const ruleStatusSavedObjectMappings: SavedObjectsType['mappings'] = {
   },
 };
 
+const truncateMessageFields = (
+  doc: SavedObjectUnsanitizedDoc<Record<string, unknown>>
+): SavedObjectSanitizedDoc => {
+  const { lastFailureMessage, lastSuccessMessage, ...restAttributes } = doc.attributes;
+
+  return {
+    ...doc,
+    attributes: {
+      lastFailureMessage: truncateMessage(lastFailureMessage),
+      lastSuccessMessage: truncateMessage(lastSuccessMessage),
+      ...restAttributes,
+    },
+    references: doc.references ?? [],
+  };
+};
+
 export const type: SavedObjectsType = {
   name: ruleStatusSavedObjectType,
   hidden: false,
   namespaceType: 'single',
   mappings: ruleStatusSavedObjectMappings,
+  migrations: {
+    '7.14.2': truncateMessageFields,
```

Comment:
If this PR doesn't end up going to 7.14.2 I'll need to adjust that.

Reply:
Looks like we missed the `7.14.2` BC, so will need to update this -- I've updated the labels here as well to target `7.15.1`.